src/Model/DICOMStructure.py

```diff
@@ -73,41 +69,37 @@ def __init__(self, patient_id, patient_name):
         """
         self.patient_id = patient_id
         self.patient_name = patient_name
-        self.studies = []
+        self.studies = {}
 
     def add_study(self, study):
         """
         Adds a Study object to the patient's list of studies.
         :param study: A Study object.
         """
-        self.studies.append(study)
+        self.studies[study.study_id] = study
 
     def has_study(self, study_id):
```

Comment:
Name of the variable is very misleading

Study ID is a DICOM element
Study Instance UID is also a DICOM element.
The docstring  indicates that this needs to be the Study Instance UID. 
This looks like there is a deeper mislabelling / use of study_id. 
If one needs to be brief, call it the study_uid, and use that in internal data classes that are representing the patient / study / series / instance hierarchy.